torch/fx/proxy.py

```diff
@@ -51,7 +51,14 @@ def create_arg(self, a: Any) -> Argument:
         """
         # aggregates
         if isinstance(a, (tuple, list)):
-            return type(a)(self.create_arg(elem) for elem in a)
+            if hasattr(a, '_fields'):
+                # NamedTuple constructors don't seem to like getting a generator
+                # expression as an argument to their constructor, so build this
+                # intermediate tuple and unpack it into the NamedTuple constructor
+                args = tuple(self.create_arg(elem) for elem in a)
+                return type(a)(*args)
```

Comment:
`create_arg` should be narrowing the set of objects allowed to those supported by `map_arg`, but NamedTuple won't be supported by map arg. Is the intension to add named tuples to map_arg? If so that should also be added, if not, then this should just erase the naming from the tuple.